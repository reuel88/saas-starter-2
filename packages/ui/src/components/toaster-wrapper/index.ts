export { ToasterWrapper } from "./toaster-wrapper";
