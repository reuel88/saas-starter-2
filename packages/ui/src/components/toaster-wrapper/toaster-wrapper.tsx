"use client";

import { Toaster } from "../ui/toaster";
import { FC, ReactNode } from "react";

type ToasterWrapperProps = {
  children: ReactNode;
};

export const ToasterWrapper: FC<ToasterWrapperProps> = ({ children }) => {
  return (
    <>
      {children}
      <Toaster />
    </>
  );
};
