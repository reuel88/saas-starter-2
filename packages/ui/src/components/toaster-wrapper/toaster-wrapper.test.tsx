import { render, screen } from "@testing-library/react";
import { ToasterWrapper } from "./toaster-wrapper";

describe("ToasterWrapper", () => {
  it("should render children and Toaster component", () => {
    // Arrange
    const children = <div>Test Children</div>;

    // Act
    const { baseElement } = render(<ToasterWrapper>{children}</ToasterWrapper>);

    // Assert
    expect(screen.getByText("Test Children")).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
