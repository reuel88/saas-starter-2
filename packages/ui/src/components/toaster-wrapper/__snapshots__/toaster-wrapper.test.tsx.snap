// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToasterWrapper > should render children and Toaster component 1`] = `
<body>
  <div>
    <div>
      Test Children
    </div>
    <div
      aria-label="Notifications (F8)"
      role="region"
      style="pointer-events: none;"
      tabindex="-1"
    >
      <ol
        class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
        tabindex="-1"
      />
    </div>
  </div>
</body>
`;
